test/redis_client/middlewares_test.rb

```diff
@@ -129,6 +129,31 @@ def test_final_errors
       TestMiddleware.calls.clear
     end
 
+    def test_final_errors_during_reconnect
```

Comment:
This test does pass on master :/

Reply:
Alright I rewrote the test to cover the connection case.

I think I also need to add one for errors that aren't connection errors.